pyfuntofem/driver/funtofem_nlbgs_driver.py

```diff
@@ -402,57 +326,26 @@ def _solve_unsteady_adjoint(self, scenario):
         for body in self.model.bodies:
             body.transfer_disps(scenario, time_index=steps)
             body.transfer_temps(scenario, time_index=steps)
-            # body.transfer_loads(scenario, time_index=steps)
 
         # Initialize the adjoint variables
         nfunctions = scenario.count_adjoint_functions()
         self._initialize_adjoint_variables(scenario, self.model.bodies)
 
         # Loop over each time step in the reverse order
-        for rstep in range(1, steps + 1):
-            step = steps - rstep + 1
+        for step in range(1, steps + 1):
+            rev_step = steps - step + 1
 
-            self.solvers.flow.set_states(scenario, self.model.bodies, step)
+            self.solvers.flow.set_states(scenario, self.model.bodies, rev_step)
             # Due to the staggering, we linearize the transfer about t_s^(n-1)
-            self.solvers.structural.set_states(scenario, self.model.bodies, step - 1)
+            self.solvers.structural.set_states(scenario, self.model.bodies, rev_step)
 
             for body in self.model.bodies:
                 body.transfer_disps_adjoint(scenario)
-                # body.transfer_loads_adjoint(scenario)
```

Comment:
Are these adjoints sensitive to the state variables of the transfer scheme? Maybe some of them are not and others are, like the load adjoint. Double check this